mw_url_rewrite.go

```diff
@@ -157,51 +153,64 @@ func urlRewrite(meta *apidef.URLRewriteMeta, r *http.Request) (string, error) {
 
 		// put url_rewrite path to context to be used in ResponseTransformMiddleware
 		ctxSetUrlRewritePath(r, meta.Path)
-
-		// matched?? Set the modified path
-		// return newpath, nil
 	}
 
+	newpath = replaceTykVariables(r, newpath, true)
+
+	return newpath, nil
+}
+
+const (
+	metaLabel    = "$tyk_meta."
+	contextLabel = "$tyk_context."
+)
+
+var dollarMatch = regexp.MustCompile(`\$tyk_context.([A-Za-z0-9_\-\.]+)`)
+var metaDollarMatch = regexp.MustCompile(`\$tyk_meta.([A-Za-z0-9_\-\.]+)`)
+
+func replaceTykVariables(r *http.Request, in string, escape bool) string {
 	contextData := ctxGetData(r)
 
-	dollarMatch := regexp.MustCompile(`\$tyk_context.([A-Za-z0-9_\-\.]+)`)
-	replace_slice := dollarMatch.FindAllStringSubmatch(rewriteToPath, -1)
-	for _, v := range replace_slice {
-		contextKey := strings.Replace(v[0], "$tyk_context.", "", 1)
+	if strings.Contains(in, contextLabel) {
+		replace_slice := dollarMatch.FindAllStringSubmatch(in, -1)
+		for _, v := range replace_slice {
+			contextKey := strings.Replace(v[0], "$tyk_context.", "", 1)
 
-		if val, ok := contextData[contextKey]; ok {
-			valStr := valToStr(val)
-			// If contains url with domain
-			if !strings.HasPrefix(valStr, "http") {
-				valStr = url.QueryEscape(valStr)
+			if val, ok := contextData[contextKey]; ok {
+				valStr := valToStr(val)
+				// If contains url with domain
+				if escape && !strings.HasPrefix(valStr, "http") {
+					valStr = url.QueryEscape(valStr)
+				}
+				in = strings.Replace(in, v[0], valStr, -1)
 			}
-			newpath = strings.Replace(newpath, v[0], valStr, -1)
 		}
 	}
 
 	// Meta data from the token
-	if session := ctxGetSession(r); session != nil {
+	session := ctxGetSession(r)
+	if session == nil {
+		return in
+	}
 
-		metaDollarMatch := regexp.MustCompile(`\$tyk_meta.(\w+)`)
-		metaReplace_slice := metaDollarMatch.FindAllStringSubmatch(rewriteToPath, -1)
+	if strings.Contains(in, metaLabel) {
```

Comment:
also could get rid of a couple of ops if we move `if session := ctxGetSession(r); ...` inside this if